libs/sdk-ui-dashboard/src/drill/DrillSelect/DrillSelectDropdown.tsx

```diff
@@ -0,0 +1,134 @@
+// (C) 2020-2021 GoodData Corporation
+
+import React from "react";
+import stringify from "json-stable-stringify";
+import { useIntl } from "react-intl";
+import invariant from "ts-invariant";
+import { IDrillEvent, UnexpectedSdkError } from "@gooddata/sdk-ui";
+import {
+    IListedDashboard,
+    isDrillToInsight,
+    isDrillToDashboard,
+    isDrillToLegacyDashboard,
+} from "@gooddata/sdk-backend-spi";
+import { Overlay } from "@gooddata/sdk-ui-kit";
+import { isDrillDownDefinition } from "@gooddata/sdk-ui-ext";
+import { IInsight, insightRef, insightTitle, ObjRef, areObjRefsEqual } from "@gooddata/sdk-model";
+import { DashboardDrillDefinition, isDrillToUrl } from "../interfaces";
+import { DrillSelectListBody } from "./DrillSelectListBody";
+import { getDrillDownAttributeTitle } from "../utils/drillDownUtils";
+import { DrillSelectContext, DrillType, DrillSelectItem } from "./interfaces";
+import { useDashboardSelector } from "../../model/state/dashboardStore";
+import { selectListedDashboards } from "../../model/state/listedDashboards/listedDashboardsSelectors";
+import { dashboardMatch } from "../utils/dashboardPredicate";
+import { selectDashboardTitle, selectInsights } from "../../model";
+
+export interface DrillSelectDropdownProps extends DrillSelectContext {
+    dropDownAnchorClass: string;
+    isOpen: boolean;
+    onClose: () => void;
+    onSelect: (item: DashboardDrillDefinition) => void;
+}
+
+export const DrillSelectDropdown: React.FC<DrillSelectDropdownProps> = (props) => {
+    const { isOpen, dropDownAnchorClass, onClose, onSelect, drillDefinitions, drillEvent } = props;
+
+    const intl = useIntl();
+    const listedDashboards = useDashboardSelector(selectListedDashboards);
+    const dashboardTitle = useDashboardSelector(selectDashboardTitle);
+    const insights = useDashboardSelector(selectInsights);
+
+    const drillSelectItems = createDrillSelectItems(
+        drillDefinitions,
+        drillEvent,
+        insights,
+        listedDashboards,
+        dashboardTitle,
+        intl.formatMessage({ id: "drill_modal_picker.more.details" }),
+    );
+
+    return isOpen ? (
+        <div className="gd-drill-modal-picker-overlay-mask">
+            <Overlay
+                closeOnOutsideClick={true}
+                closeOnEscape={true}
+                alignTo={`.${dropDownAnchorClass}`}
+                zIndex={5003}
```

Comment:
We should really have some z-index constants on some centralized place so that we can see which is used where. Feel free to do this in a follow up though